From: OpenWrt Community <openwrt@openwrt.org>
Date: Sat, 30 May 2026 00:00:00 +0000
Subject: [PATCH] arm64: dts: qcom: ipq5018: add NSS DP nodes

Add NSS Data Plane (NSS-DP) nodes required for hardware accelerated
ethernet on IPQ5018. The NSS-DP block handles packet offloading
between the network interfaces and the NSS acceleration cores.

IPQ5018 has 2 NSS-DP ports:
- Port 1: connected to internal GE PHY (GEPHY)
- Port 2: connected to UNIPHY (WAN/SGMII)

Signed-off-by: OpenWrt Community <openwrt@openwrt.org>
---
--- a/arch/arm64/boot/dts/qcom/ipq5018.dtsi
+++ b/arch/arm64/boot/dts/qcom/ipq5018.dtsi
@@ -736,6 +736,46 @@
 		};
 	};
 
+	nss-dp@3a001000 {
+		compatible = "qcom,nss-dp";
+		qcom,id = <1>;
+		reg = <0x3a001000 0x200>;
+		qcom,mactype = <2>;
+		local-mac-address = [000000000000];
+		phy-mode = "sgmii";
+	};
+
+	nss-dp@3a009000 {
+		compatible = "qcom,nss-dp";
+		qcom,id = <2>;
+		reg = <0x3a009000 0x200>;
+		qcom,mactype = <2>;
+		local-mac-address = [000000000000];
+		phy-mode = "sgmii";
+	};
+
+	nss_crypto: nss-crypto {
+		compatible = "qcom,nss-crypto-v2.0";
+		reg-names = "crypto_pbase", "crypto_sbase";
+		reg = <0x39900000 0x20000>,
+		      <0x39904000 0x1000>;
+		qcom,num-engines = <1>;
+		status = "okay";
+	};
+
